components/runners/ambient-runner/ambient_runner/bridges/claude/grpc_transport.py

```
@@ -0,0 +1,339 @@
+                stream_queue = active_streams.get(thread_id)
+                if stream_queue is not None:
+                    try:
+                        stream_queue.put_nowait(event)
+                    except asyncio.QueueFull:
+                        logger.warning(
+                            "[GRPC LISTENER] SSE tap queue full, dropping event: thread=%s",
+                            thread_id,
+                        )
+                await writer.consume(event)
+        except Exception as exc:
+            logger.error(
```

Ruff (0.15.6)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)